File: android/src/main/java/com/lightningkite/butterflytemplate/layouts/VideoDemoXml.kt
//
// VideoDemoXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.butterflytemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.butterfly.views.widget.*
import com.lightningkite.butterfly.views.*
import com.lightningkite.butterfly.android.*
import com.lightningkite.butterflytemplate.R

class VideoDemoXml {

    lateinit var video: com.lightningkite.butterfly.views.widget.VideoPlayer
    lateinit var thumbnail: ImageView
    lateinit var play: Button
    lateinit var gallery: Button
    lateinit var camera: Button
    lateinit var galleryMulti: Button
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ActivityAccess): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.video_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        video = view.findViewById<com.lightningkite.butterfly.views.widget.VideoPlayer>(R.id.video)
        thumbnail = view.findViewById<ImageView>(R.id.thumbnail)
        play = view.findViewById<Button>(R.id.play)
        gallery = view.findViewById<Button>(R.id.gallery)
        camera = view.findViewById<Button>(R.id.camera)
        galleryMulti = view.findViewById<Button>(R.id.galleryMulti)
        
        
        
        return view
    }
}

File: web/src/layout/VideoDemoXml.ts
//
// VideoDemoXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId, startupAutoResize } from 'butterfly-web/dist/views/html'
import { customViewSetDelegate } from 'butterfly-web/dist/views/CustomView'
import htmlForDefault from './video_demo.html'
//! Declares com.lightningkite.butterflytemplate.layouts.VideoDemoXml
export class VideoDemoXml {
    xmlRoot!: HTMLElement
    video!: HTMLVideoElement
    thumbnail!: HTMLImageElement
    play!: HTMLButtonElement
    gallery!: HTMLButtonElement
    camera!: HTMLButtonElement
    galleryMulti!: HTMLButtonElement
    loadHtmlString(): string {
        return htmlForDefault
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString())
        this.xmlRoot = view
        this.video = getViewById<HTMLVideoElement>(view, "video")
        this.thumbnail = getViewById<HTMLImageElement>(view, "thumbnail")
        this.play = getViewById<HTMLButtonElement>(view, "play")
        this.gallery = getViewById<HTMLButtonElement>(view, "gallery")
        this.camera = getViewById<HTMLButtonElement>(view, "camera")
        this.galleryMulti = getViewById<HTMLButtonElement>(view, "galleryMulti")
        return view
    }
}
